Sources/X509/OCSP/OCSPPolicy.swift

```diff
@@ -42,6 +37,40 @@ extension ASN1ObjectIdentifier {
     static let sha1NoSign: Self = [1, 3, 14, 3, 2, 26]
 }
 
+struct OCSPResponderSigningPolicy: VerifierPolicy {
+    /// direct issuer of the certificate for which we check the OCSP status for
+    var issuer: Certificate
+    mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+        // The root of the chain is always guaranteed to be the issuer as the root certificate store only contains the issuer
+        guard chain.last == issuer else {
+            return .failsToMeetPolicy(reason: "OCSP response must be signed by the certificate issuer or a certificate that chains up to the issuer")
+        }
+        if chain.count == 1 {
+            // the leaf is the issuer which does not need to have the OCSP signing extended key usage
+            return .meetsPolicy
+        }
+        
+        let leaf = chain.leaf
+        
+        // RFC 6960 Section 4.2.2.2. Authorized Responders
+        // OCSP signing delegation SHALL be designated by the inclusion of
+        // id-kp-OCSPSigning in an extended key usage certificate extension
+        // included in the OCSP response signer's certificate.
+        guard let extendedKeyUsage: ExtendedKeyUsage = try? leaf.extensions.extendedKeyUsage else {
+            return .failsToMeetPolicy(reason: "OCSP response certificate has no extended key usages")
+        }
+        let hasOCSPSigningUsage = extendedKeyUsage.usages.contains {
+            $0 == ExtendedKeyUsage.Usage(oid: .ExtendedKeyUsage.ocspSigning)
```

Comment:
I think this can be simplified: `ExtendedKeyUsage.Usage` is equatable, so you can use the simpler `contains` function.